agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -309,11 +312,12 @@ impl DeriveSchema for Stage {
                 .aggregations
                 .iter()
                 .map(|(k, e)| {
-                    let field_schema = e.derive_schema(state)?;
+                    let field_schema = e.derive_schema(state)?.upconvert_missing_to_null();
```

Comment:
there's a few upconvert missing to nulls sprinkled around this pr, a common mistake / fix needed for window or group funcs.